### PR TITLE
Add server argument to specify custom PAA root certificate location

### PR DESCRIPTION
**Pull request by agners**
Add a new server argument --paa-root-cert-dir to allow a custom location for PAA root certificates. This allows to store them in a location which is preserved.

### CODE REVIEW

**Comment by marcelveldt** [comment]
one lint issue

File: matter_server/server/server.py [hunk 1]
```diff
@@ -108,6 +110,16 @@ def __init__(
         self.port = port
         self.listen_addresses = listen_addresses
         self.primary_interface = primary_interface
+        if paa_root_cert_dir is None:
+            self.paa_root_cert_dir = (
+                pathlib.Path(__file__)
+                .parent.resolve()
+                .parent.resolve()
+                .parent.resolve()
+                .joinpath("credentials/development/paa-root-certs")
```
Comment (marcelveldt) [inline matter_server/server/server.py:119]: this can be a constant like DEFAULT_PAA_ROOT_CERTS_DIR

File: matter_server/server/server.py [hunk 2]
```diff
@@ -134,7 +146,7 @@ async def start(self) -> None:
         self.loop = asyncio.get_running_loop()
         self.loop.set_exception_handler(_global_loop_exception_handler)
         self.loop.set_debug(os.environ.get("PYTHONDEBUG", "") != "")
-        await self.device_controller.initialize()
+        await self.device_controller.initialize(self.paa_root_cert_dir)
```
Comment (marcelveldt) [inline matter_server/server/server.py:149]: Did you check if this actually works now ? In the past this did not work as it was hardcoded somewhere. Did you commission a new device ?

Comment (agners) [reply matter_server/server/server.py:142]: Yes, tested by commissioning a new device (WiZ), worked with an alternative path :+1: 

```
...
2024-03-07 20:55:14 (Dummy-2) CHIP_PROGRESS [chip.native.CTL] Successfully finished commissioning step 'AttestationVerification'
...
```

**Review by marcelveldt — APPROVED** [review]

**Pull request merged**